packages/sdk/wallets/wallet-keystore/src/keystores.ts

```diff
@@ -0,0 +1,227 @@
+    const addressToFile: Record<string, string> = {}
+    ;(await this.getAllKeystoreNames()).forEach(
+      (file) => (addressToFile[this.getAddress(file)] = file)
+    )
+    return addressToFile
+  }
+
+  /**
+   * Encrypts and stores a private key as a new keystore entry
+   * @param privateKey Private key to encrypted
+   * @param passphrase Secret string to encrypt private key
+   */
+  async importPrivateKey(privateKey: string, passphrase: string) {
+    // Duplicate entries for the same private key are not permitted
+    const address = normalizeAddressWith0x(privateKeyToAddress(privateKey))
+    if ((await this.listKeystoreAddresses()).includes(address)) {
+      throw new Error(ErrorMessages.KEYSTORE_ENTRY_EXISTS)
+    }
+
+    const key = Buffer.from(privateKey, 'hex')
+    const wallet = Wallet.fromPrivateKey(key)
+    const keystore = await wallet.toV3String(passphrase)
+    const keystoreName = wallet.getV3Filename(Date.now())
+
+    this.persistKeystore(keystoreName, keystore)
+  }
+
+  /**
+   * Gets name of keystore entry corresponding to an address
+   * @param address Account address
+   * @returns Name of corresponding keystore entry
+   */
+  async getKeystoreName(address: string): Promise<string> {
+    const keystoreName = (await this.getAddressMap())[normalizeAddressWith0x(address)]
+    if (keystoreName === undefined) {
+      throw new Error(ErrorMessages.NO_MATCHING_ENTRY)
+    }
+    return keystoreName
+  }
+
+  /**
+   * Gets decrypted (plaintext) private key for an account address
+   * @param address Account address
+   * @param passphrase Secret phrase used to encrypt the private key
+   * @returns
+   */
+  async getPrivateKey(address: string, passphrase: string): Promise<string> {
+    const rawKeystore = this.getRawKeystore(await this.getKeystoreName(address))
+    // TODO do we want to trim leading 0x here? what is the best practice here?
+    return (await Wallet.fromV3(rawKeystore, passphrase)).getPrivateKeyString()
+  }
+
+  /**
+   * Change secret phrase used to encrypt the private key of an address
+   * @param address Account address
+   * @param oldPassphrase Secret phrase used to encrypt the private key
+   * @param newPassphrase New secret phrase to re-encrypt the private key
+   */
+  async changeKeystorePassphrase(address: string, oldPassphrase: string, newPassphrase: string) {
+    const keystoreName = await this.getKeystoreName(address)
+    const rawKeystore = this.getRawKeystore(keystoreName)
+    const newKeystore = await (await Wallet.fromV3(rawKeystore, oldPassphrase)).toV3String(
+      newPassphrase
+    )
+    this.persistKeystore(keystoreName, newKeystore)
+  }
+
+  /**
+   * Permanently removes keystore entry from keystore
+   * @param address Account address of keystore to be deleted
+   */
+  async deleteKeystore(address: string) {
+    this.removeKeystore(await this.getKeystoreName(address))
+  }
+}
+
+export class InMemoryKeystore extends KeystoreBase {
```

Comment:
What do you think of having separate files for the subclasses?

Reply:
went back and forth on this a bit--I was nervous splitting them all up at this stage would create unnecessary clutter (esp. in the tests' imports). I'd value the "fresh-eyed" perspective here though -- do you think it would be more readable/more in line with how we prefer to structure stuff?